### PR TITLE
Allow users to pass in their own tcell.Screen to app

### PR DESCRIPTION
**Pull request by hinshun**
This is useful if the user wants to pass in their own implementation of `tcell.Screen` or a wrapper. Its also seems to make the `DontActivate` arg redundant, I imagine that is only used when the screen is managed by the user.

### CODE REVIEW

**Comment by gcla** [comment]
Hi @hinshun - will test this out this weekend :-)

**Comment by gcla** [comment]
And I'll try this one too! I don't think there'll be an issue with it. Thanks for all the PRs :-)

File: app.go [hunk 1]
```diff
@@ -275,8 +279,8 @@ func newApp(args AppArgs) (rapp *App, rerr error) {
 		log:               args.Log,
 	}
 
-	if !args.DontActivate {
-		if err = res.initScreen(); err != nil {
+	if args.Screen == nil {
+		if err := res.initScreen(); err != nil {
```
Comment (gcla) [inline app.go:283]: Hi @hinshun - should this be `args.Screen != nil` ?

**Review by gcla — CHANGES_REQUESTED** [review]
Hi @hinshun - could you check the one comment I added? Otherwise, looks good and I'll merge

File: app.go (hunk 1 above)
Comment (hinshun) [reply app.go:283]: If `args.Screen == nil` it means the user hasn't given their own screen, so `screen` should be managed by gowid. (Initialized on line 240)

Comment (gcla) [reply app.go:283]: Thanks @hinshun , you're right, I'm an idiot. I fixed up termshark to compile against your PR and found that keypresses weren't working for several seconds after the process started. But it was a bug in my (termshark's) use of the tcell.Screen!

Comment (hinshun) [reply app.go:283]: Haha, no worries. Thanks for the review!

**Comment by gcla** [comment]
Hi @hinshun - if you have time, would you mind checking that my latest commit hasn't broken your application in any way?

https://github.com/gcla/gowid/commit/072f0882aae0d0d05eaf7a9e6eed5067c26c9b13

I realized I'd broken something in termshark, but I think I have reconciled termshark's needs with your PR to allow passing in your own screen.

**Comment by hinshun** [comment]
No worries, looks like it works!

**Pull request merged**